content/news/2021-07-15-gsoc-2021-intro.md

```diff
@@ -0,0 +1,190 @@
+[@DevinR528]: https://github.com/DevinR528
+[ruma/ruma#664]: https://github.com/ruma/ruma/pull/664
+[@jplatte]: https://github.com/jplatte
+
+## What's next?
+
+I'm looking forward to implementing the MSC for [Secure Secret Storage and
+Sharing][SSSS].  This is the second large part of my project, it will involve
+adding a number of events to support storing and sharing both keys and secrets.
+To achieve full support, I would also need to implement crypto routines. I
+expect that adding these will require more effort than adding the events, though
+I look forward to the challenge.
+
+I found my first half of GSoC to be very enriching. I really enjoyed working on
+this project: I got to learn how to read and apply a specification, and I
+learned the rigour required for a project such as Ruma. I have sometimes been
+caught off-guard: sometimes I miss extraneous newlines, sometimes (often) I
+forget to add a changelog entry, and other similar more stylistic rather than
+functional issues. Ruma is a big project with many crates, yet it maintains
+consistency throughout, a quality I greatly appreciate. I now know that this
+consistency comes from the thoroughness with which each change is checked, and I
+hope to improve my own rigour when it comes to each of my pull requests.
+
+[SSSS]: https://github.com/matrix-org/matrix-doc/blob/master/proposals/1946-secure_server-side_storage.md
+
+<hr />
+
+# Ruma Tooling GSoC
+
+by Devin Ragotzy
+
+It's been a busy few weeks! I have started the process of checking four todo
+items off of [Ruma's automated checks][ruma/ruma#221] issue by contributing
+lints to the [Clippy][] project.
+
+- [check macro brace usage][rust-lang/rust-clippy#7299]
+- [check import rename][rust-lang/rust-clippy#7300]
+- [check for disallowed types][rust-lang/rust-clippy#7315]
+
+I [temporarily broke Clippy][rust-lang/rust-clippy#7434], Oh No! Luckily, the
+fix seems to have worked.
+
+- [fix the macro braces check][rust-lang/rust-clippy#7431]
+
+I've spent a lot of time familiarizing myself with the rustc [usefulness][]
+checker. The is how the Rust compiler determines if you have a match for all
+possible patterns and checks if any of the patterns are useless.
+
+```rust
+let x = true;
+match x { // `x` is matched exhaustively (all match patterns are useful)
+    true => {},
+    false => {},
+    // true => {}, would be un-useful
+}
+```
+
+This is where I now plan to add a lint to check for missed patterns on structs
+and enums that are marked `non_exhaustive` ([see issue][rust-lang/rust#84332]).
+This is helpful to Ruma because all of our public types are marked in this way,
+allowing us to update our types in a non-breaking way while still informing
+users of the change. Also, for anyone interested, I saw an interesting [blog
+post][usefulness checking blog] about usefulness checking.
+
+- [rustc reachable lint PR][rust-lang/rust#86809]
+
+For the last few days, I have been working with my mentor Jonas Platte to find
```

Comment:
I was mentioned above with my username, now with full name. Might be good to have that consistent? I don't care that much either way though.